### PR TITLE
Custom Copr builders

### PR DESCRIPTION
**Pull request by praiskup**

### CODE REVIEW

File: _posts/2024-06-22-custom-copr-builders.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+In Copr, we deploy the [Resalloc server][resalloc] to manage our farm of
+builders.  The _Copr_ build system (specifically the _copr-backend_ package)
+acts as the "client," creating tickets with _Resalloc_ and requesting specific
+kinds of worker virtual machines ("resources") for particular build tasks.
+To do this, _Copr_ uses a known set of per-ticket tags to describe particular
+needs (e.g., `--tag arch_x86_64`, `--tag powerful`, and so on).
+
+_Resalloc's_ task is to match the _client's_ ticket tags with pre-allocated VMs
+(or allocate new VMs if needed).
```
Comment (FrostyX) [inline _posts/2024-06-22-custom-copr-builders.md]: I'd probably suggest dumbing down the first two paragraphs. There is IMHO too much detail about tags, copr-backend as a client, and so on. I'd rather mention that our farm consists of builders in multiple clouds and multiple architectures. And what are the current limitations (for payoff when mentioning the feature in later paragraphs) like having to wait in the queue and pre-configured VM specs.

File: _posts/2024-06-22-custom-copr-builders.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+What's Needed?
+--------------
+
+Copr administrators should be able to configure a specific _Resalloc_ pool of
+VMs (builders) for you and make *your builds* use them.  Of course, they'll need
+your permission to start the machines in your name (_IBM Cloud_ or _EC2_ token
+needed, SSH key granting access to your VM hypervisor machine, or
+similar).
```
Comment (FrostyX) [inline _posts/2024-06-22-custom-copr-builders.md]: Can the custom builders be anywhere or does it have to be something that resalloc can interact with?
https://github.com/praiskup/resalloc?tab=readme-ov-file#what-do-you-mean-by-resources
Either way, I'd be more explicit about it.

**Review by FrostyX — COMMENTED** [review]
Thank you for the blog post :-)

File: _posts/2024-06-22-custom-copr-builders.md (hunk 1 above)
Comment (praiskup) [reply _posts/2024-06-22-custom-copr-builders.md]: Makes sense, going to drop useless theory/details!
Having to wait in queue? There's no competition with other users in this case :shrug: so users should not wait..

File: _posts/2024-06-22-custom-copr-builders.md (hunk 2 above)
Comment (praiskup) [reply _posts/2024-06-22-custom-copr-builders.md]: I dont think we have to limit this, and good idea with making it explicit 👌

E.g. using Azure would be more work for us (golden image preparation, scripting, debugging cloud issues from time to time..). And I'm curious if we (the team) want that at all (i'd vote for yes, just to learn something new, but..).

File: _posts/2024-06-22-custom-copr-builders.md (hunk 1 above)
Comment (FrostyX) [reply _posts/2024-06-22-custom-copr-builders.md]: > Having to wait in queue? There's no competition with other users in this case 🤷 so users should not wait..

I mean having to wait in queue as a normal user, and therefore motivation for somebody to add custom builders (having no competition in the queue).

Comment (praiskup) [reply _posts/2024-06-22-custom-copr-builders.md]: Thanks. Updated, marked as resolved.

File: _posts/2024-06-22-custom-copr-builders.md (hunk 2 above)
Comment (praiskup) [reply _posts/2024-06-22-custom-copr-builders.md]: Updated.

**Pull request merged**